### DISCUSSION TITLE
[REQ] add filter to all html outputs

### DISCUSSION THREAD

**Discussion by eddytsoi** [woocommerce/woocommerce#45078] [WooCommerce, Blocks, and Site Editing]
[Request]
is it possible to add more filters about html outputs, for example, the functions below:
get_image_html, get_title_html, get_rating_html, get_price_html, get_sale_badge_html, get_button_html, get_add_to_cart

**Comment by opr** [comment]
Hey @eddytsoi - thanks for starting this discussion. Just to confirm, your request is for us to create filters on the functions you mentioned, so that the HTML can be edited before being rendered?

If so, that's not a path we're likely to support, as instead we would prefer developers to use our extensibility APIs in JavaScript to modify the WooCommerce Blocks.

Could you please give some examples about what you're trying to do? Maybe we can make some suggestions about the best way to achieve it.
